### PR TITLE
Fix bug on visual prompting

### PR DESCRIPTION
**Pull request by sungchul1**
### Summary

This PR includes:
- [x] Fix bug in cascade inference that used logits with 3 dim, not 4 dim
- [x] Update default value of `is_cascade` for `infer` to True

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/models/visual_prompters/zero_shot_segment_anything.py [hunk 1]
```diff
@@ -521,15 +524,15 @@ def _predict_masks(
                 mask_input = torch.zeros(1, 1, *map(lambda x: x * 4, image_embeddings.shape[2:]), device=self.device)
                 has_mask_input = self.has_mask_inputs[0].to(self.device)
 
-            elif is_cascade and i == 1:
+            elif i == 1:
```
Comment (eunwoosh) [inline src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/models/visual_prompters/zero_shot_segment_anything.py:527]: Is it ok to execute this if `is_cascade` is False?

Comment (sungchul1) [reply src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/models/visual_prompters/zero_shot_segment_anything.py:527]: If `is_cascade` is False, `num_iter` is set to 1 at L520. It is just a redundant condition.

**Review by eunwoosh — DISMISSED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Review by eunwoosh — APPROVED** [review]
